src/pybullet_helpers/joint.py

```diff
@@ -0,0 +1,133 @@
+"""PyBullet helper class for joint utilities."""
+from typing import List, NamedTuple, Sequence, Tuple
+
+import numpy as np
+import pybullet as p
+
+from predicators.src.pybullet_helpers.geometry import Pose3D, Quaternion
+
+# Joint Positions (i.e., angles) of each joint in the body.
+# Not to be conflated with PyBullet joint states which include
+# positions, velocities and forces.
+JointPositions = List[float]
+
+
+class JointInfo(NamedTuple):
+    """Joint Information to match the output of the PyBullet getJointInfo API.
+
+    We use a NamedTuple as it supports retrieving by integer indexing
+    and most closely follows the PyBullet API.
+    """
+    jointIndex: int
+    jointName: str
+    jointType: int
+    qIndex: int
+    uIndex: int
+    flags: int
+    jointDamping: float
+    jointFriction: float
+    jointLowerLimit: float
+    jointUpperLimit: float
+    jointMaxForce: float
+    jointMaxVelocity: float
+    linkName: str
+    jointAxis: Sequence[float]
+    parentFramePos: Pose3D
+    parentFrameOrn: Quaternion
+    parentIndex: int
+
+    @property
+    def is_circular(self) -> bool:
+        """Whether the joint is circular or not."""
+        if self.is_fixed:
+            return False
+        return self.jointUpperLimit < self.jointLowerLimit
```

Comment:
I know this is from Caelan -- do you understand the logic? (I don't!) Can you add a comment if so?

Reply:
I think it's just the way Pybullet does it. Here's a continuous (circular) joint for the PR2 caster wheels: https://github.com/caelan/pybullet-planning/blob/master/models/pr2_description/pr2.urdf#L307-L317

Note you aren't supposed to specify joint limits in the URDF for continuous joints.

I debugged and on the getJointInfo and we get:

```
fl_caster_rotation_joint
lower limit: 0.0
upper limit: -1.0
```

It's probably some detail buried in the URDF parser that handles this, I'll add a comment about it